src/proto/message_translation/tbots_protobuf.h

```diff
@@ -106,6 +106,24 @@ std::unique_ptr<TbotsProto::Timestamp> createTimestamp(const Timestamp& timestam
 std::unique_ptr<TbotsProto::NamedValue> createNamedValue(const std::string name,
                                                          float value);
 
+/**
+ * Returns a TbotsProto::PlotJugglerValue proto containing the name
+ * value pairs of the map.
+ *
+ * Could use LOG(PLOTJUGGLER) to plot the values. Example:
+ *  LOG(PLOTJUGGLER) << *createPlotJugglerValue({
+ *      {"vx", velocity.x()},
+ *      {"vy", velocity.y()}
+ *  });
+ *
+ * @param values The map of name value pairs to plot
+ *
+ * @return The unique_ptr to a TbotsProto::PlotJugglerValue proto containing data with
+ *        specified names and values
+ */
+std::unique_ptr<TbotsProto::PlotJugglerValue> createPlotJugglerValue(
```

Comment:
I added this issue for making the names more consistent #2901

> also is there any limit to the number of values in the map? Could it be an issue in the future if we pass a large map, and then a large proto is made?

I don't think there is one. From my field testing, I was able to do close to 15-20 variables without any problem. So I think it should be fine.

The main reason I'm converting to Protobuf, eventhough I'm actually sending JSON to plotjuggler, is because it has a nice utility for converting maps to JSON strings. `nlohmann` is a popular JSON library used in C++ which _I think_ also allows us to go directly from `std::map` to a JSON string, but we don't currently use that library and I didn't really feel like adding more dependencies.

> Suggestions are hard limit or change signature to work with 1 key/val pair and return 1 proto at a time.

I thought about that before, but a problem I've had with being able to send only 1 value at a time as opposed to multiple is that if you have multiple variables around the same part of the code, then they each have their own timestamp (which can deviate slightly). As such, in the final graph, the graphs are slightly offsetted, instead of being aligned which is not ideal. This was a problem with `createNameValue` (Proto plotter in Thunderscope), though that also used the time when the value was received by Thunderscope, rather than when the value was sent, which made the offset much more noticeable.

https://github.com/UBC-Thunderbots/Software/blob/ae4ab422357ad77a4c0aecfc016d6c654612545a/src/proto/message_translation/tbots_protobuf.h#L106-L107

Another reason (probably the main one) is that this limits the number of UDP packets being sent over the network (groups them rather than sending them 1 by 1). I've noticed that even with the map of values, when logging from HRVO Agent in AI vs AI, it really slows down the runtime due to so many packets being sent from all of the agents.